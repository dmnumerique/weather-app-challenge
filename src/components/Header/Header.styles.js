import styled from 'styled-components'
import theme from '../../theme/theme.styles'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  background: ${theme.color.black};
  withh: 100%;
  padding: ${theme.spacing.s1} ${theme.spacing.s2};
  margin: 0 0 ${theme.spacing.s3} 0;

  @media ${theme.device.tablet} {
    flex-direction: row;
    justify-content: space-between;
  }
`

export const Heading = styled.h1`
  color: ${theme.color.white};
  font-weight: ${theme.fonts.weight.semi};
  font-size: ${theme.fonts.size.h4};
  margin: 0 0 ${theme.spacing.s2} 0;

  @media ${theme.device.tablet} {
    font-size: ${theme.fonts.size.h3};
    margin: 0;
  }
`

export const SearchWrapper = styled.div`
  display: flex;
`

export const SearchCity = styled.input`
  width: 50%;
  padding: ${theme.spacing.s1};

  @media ${theme.device.tablet} {
    min-width: 300px;
    width: auto;
  }
`

export const SearchCountry = styled.input`
  width: 50%;
  padding: ${theme.spacing.s1};

  @media ${theme.device.tablet} {
    min-width: 30px;
    width: auto;
  }
`

export const SearchButton = styled.button`
  padding: ${theme.spacing.s1} ${theme.spacing.s2};
`
