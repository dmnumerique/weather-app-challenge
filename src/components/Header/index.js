import React, { useState } from 'react'
import PropTypes from 'prop-types'
import {
  Wrapper,
  Heading,
  SearchWrapper,
  SearchCity,
  SearchCountry,
  SearchButton
} from './Header.styles'

const Header = ({ onSearch }) => {
  const [city, setCity] = useState('')
  const [country, setCountry] = useState('')

  return (
    <Wrapper>
      <Heading>Weather App</Heading>
      <SearchWrapper>
        <SearchCity
          type="text"
          name="city"
          aria-label="City name"
          placeholder="Enter city name"
          value={city}
          onChange={(event) => setCity(event.target.value)}
        />
        <SearchCountry
          type="text"
          placeholder="Enter country code"
          name="country"
          aria-label="2 digit country code"
          value={country}
          onChange={(event) => setCountry(event.target.value)}
        />
        <SearchButton type="submit" onClick={() => onSearch({ city, country })}>
          Go
        </SearchButton>
      </SearchWrapper>
    </Wrapper>
  )
}

Header.propTypes = {
  onSearch: PropTypes.func.isRequired
}

export default Header
