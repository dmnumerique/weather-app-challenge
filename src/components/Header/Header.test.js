import renderer from 'react-test-renderer'
import React from 'react'
import Header from './index'

it('should render with default props', () => {
  const tree = renderer.create(<Header onSearch={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})
