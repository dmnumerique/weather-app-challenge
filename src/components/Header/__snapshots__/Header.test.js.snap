// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #000000;
  withh: 100%;
  padding: 8px 16px;
  margin: 0 0 24px 0;
}

.c1 {
  color: #FFFFFF;
  font-weight: 600;
  font-size: 24px;
  margin: 0 0 16px 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  width: 50%;
  padding: 8px;
}

.c4 {
  width: 50%;
  padding: 8px;
}

.c5 {
  padding: 8px 16px;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 32px;
    margin: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    min-width: 300px;
    width: auto;
  }
}

@media (min-width:768px) {
  .c4 {
    min-width: 30px;
    width: auto;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Weather App
  </h1>
  <div
    className="c2"
  >
    <input
      aria-label="City name"
      className="c3"
      name="city"
      onChange={[Function]}
      placeholder="Enter city name"
      type="text"
      value=""
    />
    <input
      aria-label="2 digit country code"
      className="c4"
      name="country"
      onChange={[Function]}
      placeholder="Enter country code"
      type="text"
      value=""
    />
    <button
      className="c5"
      onClick={[Function]}
      type="submit"
    >
      Go
    </button>
  </div>
</div>
`;
